src/devices/bus/mononcol/slot.h

```diff
@@ -0,0 +1,143 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli
+#ifndef MAME_BUS_MONONCOL_SLOT_H
+#define MAME_BUS_MONONCOL_SLOT_H
+
+#pragma once
+
+#include "imagedev/cartrom.h"
+
+#include <cassert>
+
+/***************************************************************************
+ TYPE DEFINITIONS
+ ***************************************************************************/
+
+class device_mononcol_cart_interface : public device_interface
+{
+public:
+
+	// construction/destruction
+	virtual ~device_mononcol_cart_interface();
+
+	// reading and writing
+	virtual void rom_alloc(u32 size, int width, endianness_t end, char const *tag);
+
+	u8 *get_rom_base()  { return m_rom; }
+	u32 get_rom_size() { return m_rom_size; }
```

Comment:
Looking at the code, what purpose do these actually serve?  Everything’s done using the “SPI region”.  The slot can just pass in the region allocated by the software list loader, or allocate a region itself if it’s loading from a loose file.  All this does it complicate things unnecessarily.

Reply:
I don't mind

Reply:
changing this looks messy, would rather leave it for now